src/frontend/EdFi.DataManagementService.Frontend.AspNetCore/AspNetCoreFrontend.cs

```diff
@@ -55,53 +41,102 @@
     /// <summary>
     /// Converts an AspNetCore HttpRequest to a DMS FrontendRequest
     /// </summary>
-    private static async Task<FrontendRequest> FromHttpRequest(HttpRequest request)
+    private static async Task<FrontendRequest> FromRequest(HttpRequest HttpRequest, string dmsPath)
     {
         return new(
-            Body: await ExtractJsonBodyFrom(request),
-            Path: ExtractPathFrom(request),
-            QueryParameters: request.Query.ToDictionary(x => x.Key, x => x.Value[^1] ?? ""),
-            TraceId: ExtractTraceIdFrom(request)
+            Body: await ExtractJsonBodyFrom(HttpRequest),
+            Path: $"/{dmsPath}",
+            QueryParameters: HttpRequest.Query.ToDictionary(x => x.Key, x => x.Value[^1] ?? ""),
+            TraceId: ExtractTraceIdFrom(HttpRequest)
         );
     }
 
     /// <summary>
     /// Converts a DMS FrontendResponse to an AspNetCore IResult
     /// </summary>
-    private static IResult ToResult(FrontendResponse frontendResponse)
+    private static IResult ToResult(
+        FrontendResponse frontendResponse,
+        HttpContext httpContext,
+        string dmsPath
+    )
     {
-        return Results.Content(statusCode: frontendResponse.StatusCode, content: frontendResponse.Body);
+        if (frontendResponse.LocationHeaderPath != null)
+        {
+            string urlBeforeDmsPath = httpContext.Request.UrlWithPathSegment()[..^dmsPath.Length];
+            httpContext.Response.Headers.Append(
+                "Location",
+                $"{urlBeforeDmsPath}{frontendResponse.LocationHeaderPath}"
```

Comment:
I honestly don't understand this warning. I don't know if this is CodeQL being clever, or some kind of AI magic? This showed up when @bradbanister fixed the header to respond with the full resource URL (as is done in ODS API) instead of relative path.

Reply:
Breaking this down... the location header is constructed partially from the URL from the request. Kind of cool that CodeQL understands that this was, literally speaking, a user-provided URL. But maybe not a big deal, since we've already validated that the URL is meaningful. I'm going to dismiss the alert.